src/dns.ml

```diff
@@ -2873,17 +2874,17 @@ module Rr_map = struct
       else String.sub s 0 pos ^ " " ^ ws_after_56 (String.sub s pos (l - pos))
     in
     let hex cs =
-      let buf = Bytes.create (Cstruct.length cs * 2) in
-      for i = 0 to pred (Cstruct.length cs) do
-        let byte = Cstruct.get_uint8 cs i in
+      let buf = Bytes.make (String.length cs * 2) '\000' in
```

Comment:
Below exactly 2 bytes are written for each loop iteration, so `Bytes.create` should be ok too.